influence_benchmark/utils/wandb_logging.py

```diff
@@ -135,13 +154,23 @@ def extract_wandb_data(df):
 def log_iteration_data_to_wandb(turns_df, traj_df, iteration_step, top_n_trajs_per_initial_state, trajs_to_log=50):
     print(f"Logging iteration {iteration_step} to wandb")
     results = compute_iteration_statistics(traj_df, top_n_trajs_per_initial_state)
+    # This merge includes the traj-level reward/influence info to the turns_df entries for wandb logging.
+    turns_df_with_traj_rew = turns_df.merge(traj_df, on=["env_name", "initial_state_id", "trajectory_id"])
+
+    trajectories, env_stats = extract_wandb_data(turns_df_with_traj_rew, top_n_trajs_per_initial_state)
+
     current_stats = {
         "Avg reward": results["rew_avg_all_trajs"],
         "Avg reward (top n)": results["rew_avg_top_trajs"],
         "Avg influence": results["infl_avg_all_trajs"],
         "Avg influence (top n)": results["infl_avg_top_trajs"],
+        "Avg conversation length": results["length_avg_all_trajs"],
+        "Avg conversation length (top n)": results["length_avg_top_trajs"],
         "Iteration": iteration_step,
     }
+    for stat in results:
+        if "percentage" in stat:
+            current_stats[stat] = results[stat]
```

Comment:
This is kind of terrible. We should have `compute_iteration_statistics` return the list of "percentage" keys.

Reply:
Suggested a better alternative above